lib/metrics.js

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const EXPORTED_SYMBOLS = ['Metrics'];
+
+function Metrics(opts) {
+  /*
+  The metrics class is used to determine whether the recommendation is used
+  when the user navigates with the popup open. Data is sent to the Test Pilot
+  add-on, which forwards it to the Telemetry server.
+  */
+  this.events = opts.events;
+  this.interaction = null;
+  this.observerService = opts.observerService;
+  this.win = opts.win;
+  this.recommendation = opts.recommendation;
+  this.popup = opts.popup;
+  this.checked = false;
+
+  this.interactionStart = this.interactionStart.bind(this);
+  this.interactionEnd = this.interactionEnd.bind(this);
+  this.onKeyNavigate = this.onKeyNavigate.bind(this);
+  this.onClickNavigate = this.onClickNavigate.bind(this);
+}
+
+Metrics.prototype = {
+  init: function() {
+    this.win.console.log('metrics.init');
```

Comment:
We should probably consider a unified logging approach, though, to help facilitate error reporting. something like:

    function Logger (prefix, win) {
      this.classPrefix = 'universal-search';
      this.instancePrefix = prefix;
      this.win = win;
    }

    Logger.prototype = {
      _prefix: function() {
        return `[${this.classPrefix}.${this.instancePrefix}]`;
      },

      log: function() {
        [].splice.apply(arguments, [0, 0, this._prefix()]);
        this.win.console.log.apply(this.win.console, arguments);
        return this;
      }
    }


    // Then in the Metrics class...
    this.logger = new this.Logger('metrics', this.win);

    // Then in Metrics.init, to log the Metrics instance:
    this.logger.log('init', this);

    // Console sez:
    '[universal-search.metrics]', Object { ... }